attestation/sbom/sbom.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 The Witness Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sbom
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/in-toto/go-witness/attestation"
+	"github.com/in-toto/go-witness/log"
+	"github.com/invopop/jsonschema"
+)
+
+const (
+	Name    = "sbom"
+	Type    = "https://witness.dev/attestations/sbom/v0.1"
+	RunType = attestation.PostProductRunType
+
+	SPDXPredicateType      = "https://spdx.dev/Document"
+	SPDXMimeType           = "application/spdx+json"
+	CycloneDxPredicateType = "https://cyclonedx.org/bom"
+	CycloneDxMimeType      = "application/vnd.cyclonedx+json"
+)
+
+// This is a hacky way to create a compile time error in case the attestor
+// doesn't implement the expected interfaces.
+var (
+	_ attestation.Attestor = &SBOMAttestor{}
+	_ attestation.Exporter = &SBOMAttestor{}
+)
+
+func init() {
+	attestation.RegisterAttestation(Name, Type, RunType, func() attestation.Attestor {
+		return NewSBOMAttestor()
+	})
+}
+
+type Option func(*SBOMAttestor)
+
+func WithExport(export bool) Option {
+	return func(a *SBOMAttestor) {
+		a.export = export
+	}
+}
+
+type SBOMAttestor struct {
+	SBOMDocument  interface{}
```

Comment:
@jkjell do we need a json struct tag here?

Reply:
This should be handled by the `marshal()` and `unmarshal()` functions I think.